packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,197 @@
+        return this._inputs[4];
+    }
+
+    /**
+     * Gets the scale input component
+     */
+    public get scale(): NodeGeometryConnectionPoint {
+        return this._inputs[5];
+    }
+
+    /**
+     * Gets the geometry output component
+     */
+    public get output(): NodeGeometryConnectionPoint {
+        return this._outputs[0];
+    }
+
+    protected _buildBlock(state: NodeGeometryBuildState) {
+        state.pushExecutionContext(this);
+
+        this._vertexData = this.geometry.getConnectedValue(state);
+        state.pushGeometryContext(this._vertexData);
+
+        if (!this._vertexData || !this._vertexData.positions || !this.geometry.isConnected) {
+            state.restoreExecutionContext();
+            state.restoreGeometryContext();
+            this.output._storedValue = null;
+            return;
+        }
+        const count = this.count.getConnectedValue(state) as number;
+
+        if (count <= 0) {
+            this.output._storedValue = this._vertexData;
+            state.restoreExecutionContext();
+            state.restoreGeometryContext();
+        }
+
+        const origin = this.origin.getConnectedValue(state) as Vector3;
+        const direction = this.direction.getConnectedValue(state) as Vector3;
+        const rotation = this.rotation.getConnectedValue(state) as Vector3;
+        const scale = this.scale.getConnectedValue(state) as Vector3;
+
+        const invertRadians = Math.PI / 180;
+
+        const results = [];
+
+        let rootData: Nullable<VertexData> = null;
+
+        const transformMatrix = Matrix.Identity();
+        const transformOffset = Vector3.Zero();
+        const rotationOffset = Vector3.Zero();
+        const scaleOffset = Vector3.Zero();
+
+        for (this._currentIndex = 0; this._currentIndex < count; this._currentIndex++) {
+            const clone = this._vertexData.clone();
+
+            transformOffset.copyFrom(direction.clone().scale(this._currentIndex));
+            transformOffset.addInPlace(origin);
+
+            rotationOffset.copyFrom(rotation.clone().scale(this._currentIndex));
+
+            scaleOffset.copyFrom(scale.clone().scale(this._currentIndex));
+
+            scaleOffset.addInPlace(new Vector3(1, 1, 1));
+
+            Matrix.ComposeToRef(
+                scaleOffset,
+                Quaternion.FromEulerAngles(rotationOffset.x * invertRadians, rotationOffset.y * invertRadians, rotationOffset.z * invertRadians),
+                transformOffset,
+                transformMatrix
+            );
+
+            clone.transform(transformMatrix);
+
+            if (!rootData) {
+                rootData = clone;
+            } else {
+                results.push(clone);
+            }
+        }
+
+        if (rootData) {
+            rootData.merge(results, false, false, true, true);
+            this._vertexData = rootData;
+        }
+
+        // Storage
+        this.output._storedValue = this._vertexData;

```

Comment:
they should work like InstantiateBlock (with an evaluateContext property) so the user can choose to keep dynamic behavior